conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -61,13 +61,37 @@ def _get_requires_names(self, name, cpp_info):
         """
         ret = []
         for req in cpp_info.requires:
-            pkg_name, comp_name = req.split("::") if "::" in req else (name, req)
+            pkg_name, comp_name = req.split("::") if "::" in req else (dep_name, req)
             # FIXME: it could allow defining requires to not direct dependencies
             req_conanfile = self._conanfile.dependencies.host[pkg_name]
             comp_alias_name = get_component_name(req_conanfile, comp_name)
             ret.append(self._get_pc_name(pkg_name, comp_alias_name))
         return ret
 
+    def _get_requires_names(self, dep):
+        """
+        Get all the dependency's requirements (public dependencies and components)
+        """
+        # FIXME: this str(dep.ref.name) is only needed for python2.7 (unicode values).
+        #        Remove it for Conan 2.0
+        dep_name = str(dep.ref.name)
+        # At first, let's check if we have defined some component requires, e.g., "pkg::cmp1"
+        requires = self._get_component_requires_names(dep_name, dep.cpp_info)
+        # If we have found some component requires it would be enough
+        if requires:
+            return requires
+        try:
+            # If no requires were found, let's try to get all the direct
+            # dependencies, e.g., requires = "other_pkg/1.0"
+            dep_conanfile = self._conanfile.dependencies.host[dep_name]
```

Comment:
This is weird, probably `dep_conanfile == dep`. Why don't use directly `dep.direct_host.values()`? is that not working?